flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/reconciler/ReconciliationMetadata.java

```diff
@@ -42,6 +42,7 @@ public class ReconciliationMetadata {
     public static ReconciliationMetadata from(AbstractFlinkResource<?, ?> resource) {
         ObjectMeta metadata = new ObjectMeta();
         metadata.setGeneration(resource.getMetadata().getGeneration());
+        resource.getStatus().setObservedGeneration(resource.getMetadata().getGeneration());
```

Comment:
I would move this code to:
`ReconciliationStatus#serializeAndSetLastReconciledSpec` and `ReconciliationUtils#updateReconciliationMetadata` that way it's a bit more intuitive when the setting of the generation happens. It's a bit unexpected that a constructor method for `ReconciliationMetadata` mutates the resources here.

I know that my suggestion will actually duplicate the setting of the observedGeneration in the status but I think it will be cleaner once we remove the now deprecated meta generation field :)

Reply:
Done!